### ISSUE TITLE
Check Configuration -- Refresh Token -- logout not fixing -- multiple accounts

### ISSUE THREAD

**Issue by keithcdodd** [abraunegg/onedrive#1190]
**Note:** Before submitting a bug report, please ensure you are running the latest 'onedrive' client as built from 'master' and by using the latest available DMD compiler. Refer to the readme on building the client for your system.

Confirmed Version 2.4.8 (latest)

### Bug Report Details ###
**Describe the bug**
I get a repetitive error "Check your configuration as your refresh token may be empty or invalid. You may need to authorize a logout and reauthorize."

I have two accounts under two different config files. onedrivePersonal and onedriveSchool. I am confident the School one is the issue because my password changed.

I was unsure how to logout and reauthorize with multiple accounts (I found your instructions for a single account, but not multiple), but I did my best by doing:

`onedrive --confdir="~/.config/onedriveSchool" --logout`

I followed the steps by pasting the URL and then pasting back into the terminal the response URL, but I am still getting the same error.

For good measure I did the same thing for my Personal account with

`onedrive --confdir="~/.config/onedrivePersonal" --logout`

This did not fix the issue. 

**Application and Operating System Details:**
*   Provide your OS & version (CentOS 6.x, Ubuntu 18.x etc) and the output of: `uname -a`
Manjaro 20.2 "Nibia", Gnome Version 3.38.2
*   Are you using a headless system (no gui) or with a gui installed?
Did not know there is a GUI for OneDrive? That would be cool. Unless you mean like Manjaro itself... which would have a GUI I guess. Works like Windows or Mac...
*   OneDrive Account Type
Have two accounts, Personal (with personal email) and School (attached to school email)
*   Did you build from source or install from a package?
Package -- AUR onedrive-abruanegg
*   If you installed from source, what is your DMD or LDC compiler version: `dmd --version` or `ldmd2 --version`
From AUR with Manjaro, not source
*   OneDrive Application Version: Output of `onedrive --version`
`onedrive v2.4.8`
*   OneDrive Application Configuration: Output of `onedrive --display-config`
```
onedrive version                       = v2.4.8
Config path                            = /home/keith/.config/onedrive
Config file found in config path       = false
Config option 'check_nosync'           = false
Config option 'sync_dir'               = /home/keith/OneDrive
Config option 'skip_dir'               = 
Config option 'skip_file'              = ~*|.~*|*.tmp
Config option 'skip_dotfiles'          = false
Config option 'skip_symlinks'          = false
Config option 'monitor_interval'       = 300
Config option 'min_notify_changes'     = 5
Config option 'log_dir'                = /var/log/onedrive/
Config option 'classify_as_big_delete' = 1000
Config option 'upload_only'            = false
Config option 'no_remote_delete'       = false
Config option 'remove_source_files'    = false
Config option 'sync_root_files'        = false
Selective sync 'sync_list' configured  = false
Business Shared Folders configured     = false
```
*   Provide the version of curl you are using: Output of `curl --version`
```
curl 7.73.0 (x86_64-pc-linux-gnu) libcurl/7.73.0 OpenSSL/1.1.1h zlib/1.2.11 zstd/1.4.5 libidn2/2.3.0 libpsl/0.21.1 (+libidn2/2.3.0) libssh2/1.9.0 nghttp2/1.41.0
Release-Date: 2020-10-14
Protocols: dict file ftp ftps gopher http https imap imaps mqtt pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp 
Features: AsynchDNS GSS-API HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM NTLM_WB PSL SPNEGO SSL TLS-SRP UnixSockets zstd
```

*   Is your configured 'sync_dir' a local directory or a network mount point?
I don't know quite what this means. I sync to my laptop, so I assume local directory then. I have two different sync files, one for each account. onedrivePersonal and onedriveSchool.
*   If *not* local, provide all the mountpoints in your system: Output of: `mount`
Not provided, because I believe local.
*   What partition format type does your configured 'sync_dir' reside on? Output of: `lsblk -f` 
```
NAME FSTYPE FSVER LABEL UUID                                 FSAVAIL FSUSE% MOUNTPOINT
loop0
     squash 4.0                                                    0   100% /var/lib/s
loop1
     squash 4.0                                                    0   100% /var/lib/s
loop2
     squash 4.0                                                    0   100% /var/lib/s
loop3
     squash 4.0                                                    0   100% /var/lib/s
loop4
     squash 4.0                                                    0   100% /var/lib/s
loop5
     squash 4.0                                                    0   100% /var/lib/s
loop6
     squash 4.0                                                    0   100% /var/lib/s
loop7
     squash 4.0                                                    0   100% /var/lib/s
loop8
     squash 4.0                                                    0   100% /var/lib/s
loop9
     squash 4.0                                                    0   100% /var/lib/s
nvme0n1
│                                                                           
├─nvme0n1p1
│    vfat   FAT32 ESP   30DF-6832                              94.9M    49% /boot/efi
├─nvme0n1p2
│                                                                           
├─nvme0n1p3
│    BitLoc 2                                                               
├─nvme0n1p4
│    ntfs         WINRETOOLS
│                       3830AA4330AA07C8                                    
├─nvme0n1p5
│    ntfs         Image F812AA9A12AA5D7E                                    
├─nvme0n1p6
│    ntfs         DELLSUPPORT
│                       BE181DA1181D59AB                                    
└─nvme0n1p7
     ext4   1.0         de2abc25-3927-49e4-86ce-311ae4ea322c  226.8G    36% /
```

*   Explain your entire configuration setup - is the OneDrive folder shared with any other system, shared with any other platform at the same time, is the OneDrive account you use shared across multiple systems / platforms / Operating Systems and in use at the same time
OK, so I have a personal onedrive account (Personal) and a school office365 onedrive account. I sync to my laptop using this application. No other people have access to these accounts. The only other place these might sync to is to my Windows on my laptop (I have dual partition). But otherwise, it's just "in the cloud" so to speak for each account or on this laptop. I guess some other students might have access to a few shared files on their end.

**Note:** Please generate a full debug log whilst reproducing the issue as per [https://github.com/abraunegg/onedrive/wiki/Generate-debug-log-for-support](https://github.com/abraunegg/onedrive/wiki/Generate-debug-log-for-support) and email to support@mynas.com.au
There is no debug log... there is just a repetitive error (appearing about every 2 seconds) saying "Check your configuration as your refresh token may be empty or invalid. You may need to authorize a logout and reauthorize"

**To Reproduce**
Steps to reproduce the behavior if not causing an application crash:
Error repeats on it's own, every couple seconds. No steps needed. Started after school office365 account forced a new password

If issue is replicated by a specific 'file' or 'path' please archive the file and path tree & email to support@mynas.com.au 

**Complete Verbose Log Output**
A clear and full log of the problem when running the application in the following manner (ie, not in monitor mode):
```
onedrive --synchronize --verbose --syncdir="~/OneDriveAnschutz" --confdir="~/.config/onedriveAnschutz"
Using 'user' Config Dir: /home/keith/.config/onedriveAnschutz
Using 'system' Config Dir: 
No user or system config file found, using application defaults
Initializing the OneDrive API ...
Configuring Global Azure AD Endpoints
Opening the item database ...
All operations will be performed in: /home/keith/OneDriveAnschutz

ERROR: OneDrive returned an error with the following message:
  Error Message: HTTP request returned status code 400 (Bad Request)
  Error Reason:  AADSTS50173: The provided grant has expired due to it being revoked, a fresh auth token is needed. The user might have changed or reset their password. The grant was issued on '2020-10-25T23:46:24.0000000Z' and the TokensValidFrom date (before which tokens are not valid) for this user is '2020-12-13T20:26:26.0000000Z'.
Trace ID: 652370bc-71a0-40a3-94a1-5b77fa241000
Correlation ID: 690b892c-8fcb-4999-8936-ace53f674eed
Timestamp: 2020-12-18 01:19:38Z

ERROR: OneDrive returned an error with the following message:
  Error Message: HTTP request returned status code 401 (Unauthorized)
  Error Reason:  Access token is empty.

ERROR: Check your configuration as your refresh_token may be empty or invalid. You may need to issue a --logout and re-authorise this client.
```

Run the application in a separate terminal window or SSH session and provide the entire application output including the error & crash. When posing the logs, Please format log output to make it easier to read. See [https://guides.github.com/features/mastering-markdown/](https://guides.github.com/features/mastering-markdown/) for more details.

Application Log Output:
```bash
Verbose console log output goes here
```
^^I believe this would be the same thing I just posted above for the last part.

**Screenshots**

**Additional context**

### Bug Report Checklist ###
*   [yes] Detailed description
*   [yes] Application and Operating System Details provided in full
*   [yes] Reproduction steps (if applicable)
*   [yes] Verbose Log Output from your error
*   [NA? Or don't know what else to show] Debug Log generated and submitted

**Comment by abraunegg** [comment]
@keithcdodd 
Please follow the correct support proceedures as detailed here: https://github.com/abraunegg/onedrive#reporting-issues

You issue is missing:
* All relevant details as required for support
* You have not used the bug report template which requests all these details

Please fix your issue submission

**Comment by keithcdodd** [comment]
Sorry about that. I believe I fixed it. Please let me know if you need more information.

**Comment by abraunegg** [comment]
@keithcdodd 
> I have two accounts under two different config files. onedrivePersonal and onedriveSchool. I am confident the School one is the issue because my password changed.

First, stop any systemd onedrive service process that *may* be running the client in an automatic fashion. You do not detail how you are running the application normally when this error is not occurring.

Once you have confirmed all systemd / application processes have been stopped, please try this command:
```
onedrive --confdir="~/.config/onedriveAnschutz" --logout
```
This *should* remove the required files and require you to re-authenticate.

Once re-authenticated, please do the following command:
```
ls -la ~/.config/onedriveAnschutz
```
This will list all the files in the configuration directory and their permissions. Please provide this so this can be confirmed as OK.

Once this is done, *test* the client with the updated authentication:
```
onedrive --confdir="~/.config/onedriveAnschutz" --synchronize --verbose --dry-run
```

If this is all OK, then you can enable any systemd service process again.

**Comment by keithcdodd** [comment]
Thank you. I don't know why, but I struggled to find my system services... but finally in `/usr/lib/systemd/user` I found `onedrive-anschutz` so I ran the command `systemctl --user stop onedrive-anschutz` before doing the suggested `onedrive --confdir="~/.config/onedriveAnschutz" --logout`.

Anyways, after that, the output of `ls -la ~/.config/onedriveAnschutz` is:
`total 46584
drwxr-xr-x  2 keith keith     4096 Dec 19 12:58 .
drwxr-xr-x 54 keith keith     4096 Dec 17 18:21 ..
-rw-r--r--  1 keith keith    98304 Dec 19 12:56 items.sqlite3-shm
-rw-r--r--  1 keith keith 47586032 Dec 19 12:56 items.sqlite3-wal
-rw-------  1 keith keith     1141 Dec 19 12:58 refresh_token
`
Testing with `onedrive --confdir="~/.config/onedriveAnschutz" --synchronize --verbose --dry-run` seemed to go well, so then I started it again with:
`systemctl --user start onedrive-anschutz`

so now I hope/think we are good.

Thank you so much! The hardest part for me was pinning down the systemd service... I really appreciate the help.
